react/week3/todolist/my-app/src/components/TodoItems.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import WatchCount from './WatchCount';
+import Todo from './Todo';
+import TodoForm from './TodoForm';
+import uuid from 'uuid';
+import LoaderComponent from './LoaderComponent';
+//import DatePicker from 'react-date-picker';
+
+const TodoItems = () => {
+	const [items, setItems] = useState([]);
+	useEffect(() => {
+		(async () => {
+			const todosList = await fetch(
+				'https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw'
+			).then(response => response.json());
+			setItems(todosList);
+		})();
+	}, []);
+	const addItem = (text, date) => {
+		setItems([
+			...items,
+			{
+				id: uuid(),
+				description: text,
+				deadline: date,
+				completed: false,
+				isEditMode: false,
+			},
+		]);
+	};
+	const handleCheckedStatus = id => {
+		setItems(prevState => {
+			const updatedTodos = prevState.map(todo => {
+				if (todo.id === id) {
+					return { ...todo, completed: !todo.completed };
+				}
+				return todo;
+			});
+			return updatedTodos;
+		});
+	};
+	const handleDelete = id => {
+		setItems(prevState => {
+			const updatedTodoList = prevState.filter(todo => todo.id !== id);
+			return updatedTodoList;
+		});
+	};
+	const handleEdit = id => {
+		setItems(prevState => {
+			const updatedTodos = prevState.map(todo => {
+				if (todo.id === id) {
+					return { ...todo, isEditMode: !todo.isEditMode };
+				}
+				return todo;
+			});
+			return updatedTodos;
+		});
+	};
+
+	const handleUpdate = (id, value) => {
+		setItems(prevState => {
+			const updatedTodos = prevState.map(todo => {
+				if (todo.id === id) {
+					return { ...todo, description: value };
+				}
+				return todo;
+			});
+			return updatedTodos;
+		});
+	};
+
+	return (
+		<div>
+			<WatchCount />
+			<TodoForm addTodo={addItem} />
+			{items !== null ? <ul>{items.length === 0 ? 'No Items' :
```

Comment:
It doesn't look like `items` can ever be `null`

Reply:
Hi @Hatko,
Thank you :)
I have used it for the LoaderComponent,I  just to show , when the items are being fetch.